pkg/agent/openflow/network_policy.go

```diff
@@ -904,24 +959,37 @@ func (c *client) BatchInstallPolicyRuleFlows(ofPolicyRules []*types.PolicyRule)
 	c.replayMutex.RLock()
 	defer c.replayMutex.RUnlock()
 
-	var allCtxChanges []*conjMatchFlowContextChange
 	var allFlows []binding.Flow
-	var updatedConjunctions []*policyRuleConjunction
+	var conjunctions []*policyRuleConjunction
 
 	for _, rule := range ofPolicyRules {
 		conj := c.calculateActionFlowChangesForRule(rule)
-		ctxChanges := c.calculateMatchFlowChangesForRule(conj, rule, true)
+		c.addRuleToConjunctiveMatch(conj, rule)
 		allFlows = append(allFlows, conj.actionFlows...)
 		allFlows = append(allFlows, conj.metricFlows...)
-		allCtxChanges = append(allCtxChanges, ctxChanges...)
-		updatedConjunctions = append(updatedConjunctions, conj)
+		conjunctions = append(conjunctions, conj)
+	}
+
+	for _, ctx := range c.globalConjMatchFlowCache {
+		actions := make([]*conjunctiveAction, 0, len(ctx.actions))
+		for _, action := range ctx.actions {
+			actions = append(actions, action)
+		}
+		ctx.flow = c.conjunctiveMatchFlow(ctx.tableID, ctx.matchKey, ctx.matchValue, ctx.priority, actions)
+		allFlows = append(allFlows, ctx.flow)
+		if ctx.dropFlow != nil {
+			allFlows = append(allFlows, ctx.dropFlow)
+		}
 	}
+
 	// Send the changed Openflow entries to the OVS bridge.
-	if err := c.sendConjunctiveFlows(allCtxChanges, allFlows); err != nil {
+	if err := c.ofEntryOperations.AddAll(allFlows); err != nil {
+		// Discard the cache if the bundle fails.
+		c.globalConjMatchFlowCache = map[string]*conjMatchFlowContext{}
```

Comment:
Yes, we only use the method when agent starts. Will add comment about it.
Actually we already have a bug because we don't clean the cache, see the fix here: https://github.com/antrea-io/antrea/pull/2465.

Reply:
added comment